tests/test_slim_engine.py

```diff
@@ -315,6 +315,65 @@ def test_simulate(self):
         assert observed_counts == [0, 0, 8]
         assert all(tree.num_roots == 1 for tree in ts.trees())
 
+    def test_simulate_dfe(self):
+        slim_path = os.environ.get("SLIM", "slim")
+
+        def count_mut_types(ts):
+            mut_info = {}
+            for mut in ts.mutations():
+                for j, md in zip(
+                    mut.derived_state.split(","), mut.metadata["mutation_list"]
+                ):
+                    if j not in mut_info:
+                        mut_info[int(j)] = md
+            num_neutral = sum([mut_info[j]["selection_coeff"] == 0.0 for j in mut_info])
+            return [num_neutral, abs(len(mut_info) - num_neutral)]
+
+        # test a DFE without demography
+        with tempfile.NamedTemporaryFile(mode="w") as f:
+            cmd = (
+                f"-q -e slim --slim-scaling-factor 20 --slim-path {slim_path} "
+                f"HomSap -c chr22 -l0.001 -o {f.name} --dfe Gamma_K17 10"
+            ).split()
+            capture_output(stdpopsim.cli.stdpopsim_main, cmd)
+            ts = tskit.load(f.name)
+        assert ts.num_samples == 10
+        assert all(tree.num_roots == 1 for tree in ts.trees())
+        n_mut_types = count_mut_types(ts)
+        assert n_mut_types[0] > 0
+        assert n_mut_types[1] > 0
+
+        # test a DFE without demography with interval
+        with tempfile.NamedTemporaryFile(mode="w") as f:
+            cmd = (
+                f"-q -e slim --slim-scaling-factor 20 --slim-path {slim_path} "
+                f"HomSap -c chr22 -l0.001 -o {f.name} --dfe Gamma_K17 "
+                "--interval [[1000,2000]] 10"
```

Comment:
does this work? I'd thought you'd need quotes around the interval.

Reply:
i mean the test pass and non-neutral sites are observed, so I guess it works?

Reply:
oh! That's because there's no space in `[[1000,2000]]`! =)